drivers/regulator/CMakeLists.txt

```diff
@@ -1,6 +1,4 @@
 # Copyright 2020 Peter Bigot Consulting, LLC
 # SPDX-License-Identifier: Apache-2.0
 
-zephyr_library()
```

Comment:
Every other driver subsystem puts its sources into a library.  This one should too when another type of regulator is added.  So are we just kicking the problem down the road?

Also, the next line adds `regulator_fixed.c` as a library source.  What library is it going into?

Reply:
> Every other driver subsystem puts its sources into a library.

No, this is not consistent. I will think it over again.


> Also, the next line adds `regulator_fixed.c` as a library source. What library is it going into?

my mistake, this should probably be `zephyr_sources_ifdef(CONFIG_REGULATOR_FIXED regulator_fixed.c`) in that case.

Reply:
@tejlmand the fact that enabling `CONFIG_REGULATOR` in `tests/drivers/regulator/fixed/` makes that test fail for boards that do not have a regulator in their Devicetree is what triggers the problem with `zephyr_library()`.

Reply:
> Every other driver subsystem puts its sources into a library. 

Unfortunately not, this is very inconsistent among Zephyr driver libraries.

One root cause is probable that driver libs in general don't use depends (ask someone else why they don't as I don't know) but the sources added to the driver lib uses `zephyr_[library]_sources_ifdef()` which means that if `zephyr_library()` was used, then it could end up being empty and thus result in CMake failures.

And thus people take the easy but dirty solution of just placing it `zephyr_sources()` resulting in this mess instead.
https://github.com/zephyrproject-rtos/zephyr/issues/8825

Reply:
@jfischer-no I would prefer we keep `zephyr_library()` to avoid making more mess as described in #8825.

Looking into how this can be accomplished.

Reply:
I agree on zephyr_library(). I reverted this changes and add comment to tests/drivers/regulator/fixed/prj.conf